pkg/cli/plugins_test.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cli
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"sigs.k8s.io/kubebuilder/pkg/plugin"
+)
+
+var _ = Describe("resolvePluginsByKey", func() {
+
+	var (
+		plugins = makePluginsForKeys(
+			"foo.example.com/v1.0",
+			"bar.example.com/v1.0",
+			"baz.example.com/v1.0",
+			"foo.kubebuilder.io/v1.0",
+			"foo.kubebuilder.io/v2.0",
+			"bar.kubebuilder.io/v1.0",
+			"bar.kubebuilder.io/v2.0",
+		)
+		resolvedPlugins []plugin.Base
+		err             error
+	)
+
+	It("should resolve keys correctly", func() {
```

Comment:
I think it would be easier to understand if we have 1 IT for scenario such as: 

```
It("should work when the plugin informed exist", func() {
		By("resolving foo.example.com/v1.0")
		resolvedPlugins, err = resolvePluginsByKey(plugins, "foo.example.com/v1.0")
		Expect(err).NotTo(HaveOccurred())
		Expect(makePluginKeySlice(resolvedPlugins...)).To(Equal([]string{"foo.example.com/v1.0"}))
	})
```

Reply:
Using multiple `By()`'s within an `It()` is idiomatic, check out [the docs](https://onsi.github.io/ginkgo/).

Reply:
I am not sure if here, it is so clear as the examples in the doc but ok.